node/src/components/small_network.rs

```diff
@@ -953,6 +884,67 @@ where
                     Effects::new()
                 }
             }
+            Event::ContractRuntimeAnnouncement(
+                ContractRuntimeAnnouncement::LinearChainBlock(_)
+                | ContractRuntimeAnnouncement::StepSuccess { .. },
+            ) => Effects::new(),
+            Event::ContractRuntimeAnnouncement(
+                ContractRuntimeAnnouncement::UpcomingEraValidators {
+                    era_that_is_ending,
+                    mut upcoming_era_validators,
+                },
+            ) => {
+                if era_that_is_ending < self.active_era {
+                    debug!("ignoring past era end announcement");
+                } else if era_that_is_ending != self.active_era {
+                    debug!("received future era end announcement, clearing limiters");
+                    self.active_era = era_that_is_ending + 1;
```

Comment:
Less confusing to me (and maybe others?) if we set this *after* getting the active/upcoming era validators from the map.

Reply:
I'd need a good name for a variable `era_that_is_ending + 1` then, do you have a suggestion?

Reply:
Obsolete, as the code has been rewritten in c5720c5